modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/snowplow/RedirectAdapter.scala

```diff
@@ -96,6 +96,16 @@ object RedirectAdapter extends Adapter {
               msg
             )
           Monad[F].pure(failure.invalidNel)
+        case Some(null) =>
```

Comment:
Actually, I think:

1. We should not distinguish between `None` and `Some(null)`.
2. We need to check this pattern `originalParams.get(x)` in all other adapters. Can you do that, @benjben?

Reply:
> We should not distinguish between None and Some(null)

`None` is for when `u` param is absent, `Some(null)` is for `u` exists but doesn't have a value (`&u`). Don't we want to distinguish the 2 cases in the error message to help to fix the tracking?

> Can you do that, @benjben?

Sure

Reply:
> Don't we want to distinguish the 2 cases in the error message to help to fix the tracking?

In that case, we at least should change `toMap` signature to `Map[String, Option[String]]`, but I personally don't think that it such a big deal to be specific - it should be easy to debug with payload anyway. Just `Some(null)` is such a horrible code smell.

Reply:
> it should be easy to debug with payload anyway

You're right we have the raw payload with the raw query string anyway. I'll remove this `Some(null)`